src/commands/appSettings/connectionSettings/netherite/EventHubsConnectionListStep.ts

```diff
@@ -57,20 +60,25 @@ export class EventHubsConnectionListStep<T extends INetheriteConnectionWizardCon
 
         switch (context.eventHubsConnectionType) {
             case ConnectionType.Azure:
-                if (!(context as INetheriteAzureConnectionWizardContext).subscriptionId) {
-                    Object.assign(context, createSubscriptionContext(await subscriptionExperience(context, ext.rgApiV2.resources.azureResourceTreeDataProvider)));
+                const azureContext = context as INetheriteAzureConnectionWizardContext;
+                if (!azureContext.subscriptionId) {
+                    Object.assign(azureContext, createSubscriptionContext(await subscriptionExperience(azureContext, ext.rgApiV2.resources.azureResourceTreeDataProvider)));
                 }
 
-                LocationListStep.addProviderForFiltering(context as unknown as ILocationWizardContext, EventHubsProvider, EventHubsNamespaceResourceType);
+                // Make sure this is added ahead of time because location might get prompted during `ResourceGroupListStep`
+                LocationListStep.addProviderForFiltering(azureContext as unknown as ILocationWizardContext, EventHubsProvider, EventHubsNamespaceResourceType);
 
                 promptSteps.push(
                     new ResourceGroupListStep() as AzureWizardPromptStep<INetheriteAzureConnectionWizardContext>,
                     new EventHubsNamespaceListStep(),
+                    new EventHubsNamespaceAuthRuleListStep(),
```

Comment:
Moving these steps up one level to make `EventHubsNamespaceListStep` more reuseable